docs/base/maven_plugin.rst

```diff
@@ -0,0 +1,66 @@
+.. _craft_parts_maven_plugin:
+
+Maven plugin
+============
+
+The Maven plugin builds Java projects using the Maven build tool.
+
+After a successful build, this plugin will:
+
+* Create ``bin/`` and ``jar/`` directories in ``$CRAFT_PART_INSTALL``.
+* Find the ``java`` executable provided by the part and link it as
+  ``$CRAFT_PART_INSTALL/bin/java``.
+* Hard link the ``.jar`` files generated in ``$CRAFT_PART_SOURCE`` to
+  ``$CRAFT_PART_INSTALL/jar``.
+
+
+Keywords
+--------
+
+In addition to the common :ref:`plugin <part-properties-plugin>` and
+:ref:`sources <part-properties-sources>` keywords, this plugin provides the following
+plugin-specific keywords:
+
+maven-parameters
+~~~~~~~~~~~~~~~~
+**Type:** list of strings
+
+Used to add additional parameters to the ``mvn package`` command line.
+
+
+Environment variables
+---------------------
+
+This plugin reads the ``http_proxy`` and ``https_proxy`` variables from the environment
+to configure Maven proxy access. A comma-separated list of hosts that should not be
+accessed via proxy is read from the ``no_proxy`` environment variable.
+
+Please refer to `Configuring Apache Maven <https://maven.apache.org/configure.html>`_ for
+a list of environment variables used to configure Maven.
+
+
+Dependencies
+------------
+
+The plugin expects Maven to be available on the system as the ``mvn`` executable, unless
+a part named ``maven-deps`` is defined. In this case, the plugin will assume that this
+part will stage the ``mvn`` executable to be used in the build step.
+
+A Java Runtime must be staged if not part of the execution environment.
+
+
+Example
+-------
+
+This is an example of a Snapcraft part using the Maven plugin. Note that the Maven and
```

Comment:
I think this Snapcraft-specific example should be in Snapcraft's docs, and then Rockcraft should have an example too (I think this is what we should do for all plugins)

Reply:
Moved the example to Snapcraft docs.